vnext/Shared/Shared.vcxitems

```diff
@@ -15,6 +15,105 @@
     <ProjectCapability Include="SourceItemsFromImports" />
   </ItemGroup>
   <ItemGroup>
+    <ClCompile Include="$(MSBuildThisFileDirectory)..\codegen\react\components\rnwcore\EventEmitters.cpp" DisableSpecificWarnings="4715;%(DisableSpecificWarnings)">
```

Comment:
feels like this is the dependency is going in the wrong direction; MRN depends on Shared, shouldn't be the other way around. Perhaps there is code in Shared that needs to be moved out for this to make more sense?

Reply:
Could you clarify? These files are from Meta's RN core, not MS.RN.

Reply:
@vmoroz What files are Meta's RN core? Is `codegen\react\components\rnwcore\EventEmitters.cpp` one of them?

Reply:
@asklar Unfortunately, the circular dependency already exists.
Example (Shared.vcxitems at current main):
```xml
    <ClCompile Include="$(ReactNativeWindowsDir)Microsoft.ReactNative\Fabric\DWriteHelpers.cpp" />
    <ClCompile Include="$(ReactNativeWindowsDir)Microsoft.ReactNative\Fabric\FabricUIManagerModule.cpp" />
```

We probably need a new intermediate shared items project to break this loop, but that seems to me out of the scope of this PR.

Reply:
Looks like the files were moved out of MRN into Shared by #9891, so maybe @acoates-ms can answer Alex's question.

Reply:
@JunielKatarn , you are right. This is a RNW file. I got confused with the 'core' part.
IMHO, the Shared Project must be deleted or transformed to something else:
- The name is not descriptive and rather confusing.
- I would rather see a project name that groups files for their true purpose such as "RN.Fabric", "RN.Common", "RN.Networking", etc.

Reply:
The project was originally created to common sources that couldn't be compiled in the same static library produced by project `Common.vcxproj`.

I also agree we could split it into multiple shared items projects for specialized code, but I still think there ought to be a "Common shared items" project, even if we rename it.

Note, while the filename is `Shared.vcxitems`, the actual project name in Visual Studio is shown as `Microsoft.ReactNative.Shared`.
Perhaps it could be renamed to `Microsoft.ReactNative.SharedItems`.

Reply:
The shared project isn't an actual project, right?  It's just a list of files that should be in both MS.RN.dll and rnwin32.dll? I hadn't been giving much though on project layout beyond that to be honest.  

`codegen\react\components\rnwcore\EventEmitters.cpp` is not quite part of core.  That file is codegen'd from all the components that have codegen enabled.  So, ours will have different components/properties than code does as we have forked and added components on the JS side.

Reply:
> The shared project isn't an actual project, right? It's just a list of files that should be in both MS.RN.dll and rnwin32.dll?

Yes, it is just a collection of files that can be reused between different projects. As we move towards MS.RN.Common DLL we may consider restructuring the set of project and shared project to better reflect the new targets.